src/data-access/repositories/geolocation-repository.ts

```diff
@@ -0,0 +1,37 @@
+import { GeolocationData, Maybe } from '../../graphql/types/schema-types';
+import { GeocodeResponse, GeocodeService } from '@mapbox/mapbox-sdk/services/geocoding';
+import { ApolloError } from 'apollo-server-errors';
+
+export interface GeolocationRepository {
+	getGeolocation(query: string): Promise<Maybe<GeolocationData>>;
+}
+
+export class MapboxGeolocationRepository implements GeolocationRepository {
+	constructor(private client: GeocodeService) {}
+
+	async getGeolocation(query: string): Promise<Maybe<GeolocationData>> {
+		const request = this.client.forwardGeocode({
+			query,
+			mode: 'mapbox.places',
+			limit: 1,
+		});
+
+		const response = await request.send();
+		if (response.statusCode != 200) {
+			throw new ApolloError('Error while fetching geolocation data', 'GeolocationApiRequestFailed', {
+				respoonseCode: response.statusCode,
+				respoonseBody: response.body,
```

Comment:
Just curious, is there a typo here?

Reply:
Oh, my bad! That is indeed a typo. I can fix it right away.